models/headers-1.0.0.yaml

```
@@ -0,0 +1,475 @@
+      **200** or **201** with resource (representation) in the response body on
+      success.
+    * **wait=<delta-seconds>** is used to suggest a maximum time the server has
+      time to process the request synchronously.
+    * **handling=<strict|lenient>** is used to suggest the server to be strict
+      and report error conditions or lenient, i.e. robust and try to continue,
+      if possible.
+
+    See [RFC 7340][rfc-7240] as well as [API Guideline Rule #181][api-181]
+    for further details.
+
+    [rfc-7240]: <https://tools.ietf.org/html/rfc7240>
+    [api-181]: <https://opensource.zalando.com/restful-api-guidelines/#181>
+  schema:
+    type: array
+    items:
+      type: string
+  style: simple
+  explode: false
+  example: "respond-async", "return=minimal", "wait=20", "handling=strict"
+  
+
+Idempotency-Key:
+  name: Idempotency-Key
+  in: header
+  required: false
+  description: |-
+    The **Idempotency-Key** is a free identifier created by the client to
+    identify a request. It is used by the service to identify repeated request
+    to ensure idempotent behavior by sending the same (or a similar) response
+    without executing the request a second time.
+
+    Clients should be careful as any subsequent requests with the same key may
+    return the same response without further check. Thus, it is recommended to
+    use a UUID version 4 (random) or any other random string with enough
+    entropy to avoid collisions.
+
+    Keys expire after 24 hours. Clients are responsible to stay within this
+    limit, if they require idempotent behavior.
+
+    See [API Guideline Rule #181][api-230] for further details.
+
+    [api-230]: <https://opensource.zalando.com/restful-api-guidelines/#230>
+  schema:
+    type: string
+    format: uuid
+  example: "7da7a728-f910-11e6-942a-68f728c1ba70"
+
+
+# Standard response headers
+
+
+Content-Encoding:
+  name: Content-Encoding
+  in: header
+  required: false
+  description: |-
+    The **Content-Encoding** header lists any encoding (usually a compression
+    algorithm) applied to the message payload - in execution order. This allows
+    the recipient to decode the content in order to obtain the original payload
+    format.
+
+    Supported compression algorithms of the server are [**gzip**][gzip] and
+    **identity**, but other encoding from list below may be support too:
+
+    * **gzip** - a compression format that uses [Lempel-Ziv coding][gzip]
+      (LZ77) with a 32-bit CRC.
+    * **compress** - a format using the [Lempel-Ziv-Welch][lzw] (LZW)
+      algorithm.
+    * **deflate** - A compression format that uses the [zlib][zlib] structure
+      with the [deflate][deflate] compression algorithm.
+    * **br** - a compression format that uses the [Brotli][brotli] algorithm.
+    * **identity** - the identity function without modification or compression
+      used to indicate unchanged responses. This value is always considered as
+      acceptable, even if omitted.
```

```suggestion
    * **gzip** - a[compression format with the [gzip file format][gzip]
        (including a 32 bit CRC)  around the [deflate][deflate] compression algorithm.
    * **deflate** - A compression format that uses the [zlib][zlib] structure
      with the [deflate][deflate] compression algorithm.
    * **compress** - a format using the [Lempel-Ziv-Welch][lzw] (LZW)
      algorithm.
    * **br** - a compression format that uses the [Brotli][brotli] algorithm.
    * **identity** - the identity function without modification or compression
      used to indicate unchanged responses. This value is always considered as
      acceptable, even if omitted.
```
Both gzip and "deflate" (aka zlib) are using the same algorithm (deflate), just with different wrappers.